dialogplus/src/main/java/com/orhanobut/dialogplus/DialogPlus.java

```diff
@@ -463,13 +478,16 @@ private void onAttached(View view) {
         contentContainer.startAnimation(inAnim);
 
         contentContainer.requestFocus();
-        holder.setOnKeyListener(new View.OnKeyListener() {
+            holder.setOnKeyListener(new View.OnKeyListener() {
             @Override
             public boolean onKey(View v, int keyCode, KeyEvent event) {
                 switch (event.getAction()) {
                     case KeyEvent.ACTION_UP:
                         if (keyCode == KeyEvent.KEYCODE_BACK && isCancelable) {
                             dismiss();
+                            if (onCancelListener != null) {
```

Comment:
can you also move dismiss() to the end, I think as a logic, cancel listener should be invoked before dismissing.